utils/systemd/monerod.service

```diff
@@ -8,15 +8,13 @@ Group=monero
 WorkingDirectory=~
 
 Type=forking
-ExecStart=/usr/bin/monerod --config-file /etc/monerod.conf --detach
+PIDFile=/var/run/monerod.pid
 
-# This is necessary because monerod does not yet support
-# writing a PID file, which means systemd tries to guess the PID
-# by default, but it guesses wrong (sometimes, depending on
-# random timing of events), because the daemon forks twice.
-# The ultimate fix is for the daemon to write a PID file, and
-# a workaround is to disable the guessing feature in systemd.
-GuessMainPID=no
+ExecStart=/usr/bin/monerod --config-file /etc/monerod.conf \
+    --detach --pidfile /var/run/monerod.pid
+
+# Delete the PID file after the daemon is stopped.
+ExecStopPost=/bin/rm -f /var/run/monerod.pid
```

Comment:
Relying on the service file to delete the pid file has a couple of issues. 1) this is not cross platform. On OSX I use a LauchDaemon not systemd service for example. 2) this doesn't solve for a crash of monerod or manual stop (so not stopped by service).